x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/queries/InterceptedInferenceQueryBuilder.java

```diff
@@ -321,29 +342,48 @@ private QueryBuilder doRewriteGetInferenceResults(QueryRewriteContext queryRewri
             );
         }
 
+        if (inferenceResultsMapSupplier != null) {
+            // Additional inference results have already been requested, and we are waiting for them to continue the rewrite process
```

Comment:
Would it make sense to extract the content of this if block into its own method?